rewrite-maven/src/main/java/org/openrewrite/maven/RenamePropertyKey.java

```diff
@@ -73,6 +73,7 @@ public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
                 Xml.Tag t = super.visitTag(tag, ctx);
                 if (isPropertyTag() && oldKey.equals(t.getName())) {
                     t = t.withName(newKey);
+                    maybeUpdateModel();
```

Comment:
The weird thing is that if I remove this line, the test seems to succeed anyway.

Not sure - when I run tests in gradle on Intellij a lot of chaos seems to happen. But I did see the new test succeed (with and without this line). And I can verify the original issue definitely reproduces with rewrite-maven-plugin:4.43.0.

Reply:
I just verified that **this change fixes the issue**.
The just doesn't fail if this line isn't there - because the test environment differs from reality somehow.